File: frontend/src/types/clinic.ts
export interface Clinic {
  id: string;
  name: string;
  legal_name?: string | null;
  registration_document?: string | null;
  contact_email?: string | null;
  contact_phone?: string | null;
  address_line1?: string | null;
  address_line2?: string | null;
  city?: string | null;
  state?: string | null;
  postal_code?: string | null;
  notes?: string | null;
  created_at: string;
  updated_at: string;
}

export interface ClinicCreatePayload {
  name: string;
  legal_name?: string;
  registration_document?: string;
  contact_email?: string;
  contact_phone?: string;
  address_line1?: string;
  address_line2?: string;
  city?: string;
  state?: string;
  postal_code?: string;
  notes?: string;
}

export interface ClinicUpdatePayload {
  name?: string;
  legal_name?: string;
  registration_document?: string;
  contact_email?: string;
  contact_phone?: string;
  address_line1?: string;
  address_line2?: string;
  city?: string;
  state?: string;
  postal_code?: string;
  notes?: string;
}

export interface ClinicMember {
  id: string;
  clinic_id: string;
  user_id: string;
  role: string;
  created_at: string;
  user_display_name: string;
  user_email: string;
}

export interface ClinicInvitation {
  id: string;
  clinic_id: string;
  inviter_user_id: string;
  email: string;
  role: string;
  status: string;
  clerk_invitation_id?: string | null;
  expires_at?: string | null;
  accepted_at?: string | null;
  created_at: string;
}

export interface ClinicInvitationCreatePayload {
  email: string;
  redirect_url?: string;
}

export interface ClinicInvitationResendPayload {
  redirect_url?: string;
}

File: backend/app/schemas/clinic.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, EmailStr


class ClinicFields(BaseModel):
    name: str
    legal_name: str | None = None
    registration_document: str | None = None
    contact_email: EmailStr | None = None
    contact_phone: str | None = None
    address_line1: str | None = None
    address_line2: str | None = None
    city: str | None = None
    state: str | None = None
    postal_code: str | None = None
    notes: str | None = None


class ClinicCreate(ClinicFields):
    pass


class ClinicUpdate(BaseModel):
    name: str | None = None
    legal_name: str | None = None
    registration_document: str | None = None
    contact_email: EmailStr | None = None
    contact_phone: str | None = None
    address_line1: str | None = None
    address_line2: str | None = None
    city: str | None = None
    state: str | None = None
    postal_code: str | None = None
    notes: str | None = None


class ClinicResponse(ClinicFields):
    id: UUID
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class ClinicMemberResponse(BaseModel):
    id: UUID
    clinic_id: UUID
    user_id: UUID
    role: str
    created_at: datetime
    user_display_name: str
    user_email: EmailStr


class ClinicInvitationCreate(BaseModel):
    email: EmailStr
    redirect_url: str | None = None


class ClinicInvitationResend(BaseModel):
    redirect_url: str | None = None


class ClinicInvitationResponse(BaseModel):
    id: UUID
    clinic_id: UUID
    inviter_user_id: UUID
    email: EmailStr
    role: str
    status: str
    clerk_invitation_id: str | None = None
    expires_at: datetime | None = None
    accepted_at: datetime | None = None
    created_at: datetime

    class Config:
        from_attributes = True
